compiler/rustc_typeck/src/check/compare_method.rs

```diff
@@ -929,6 +933,77 @@ fn compare_synthetic_generics<'tcx>(
     if error_found { Err(ErrorReported) } else { Ok(()) }
 }
 
+fn compare_const_generic_types<'tcx>(
```

Comment:
nit
```suggestion
fn compare_const_param_types<'tcx>(
```
or `parameter`